pkg/systemd/hwinfo.jsx

```diff
@@ -65,20 +68,115 @@ const SystemInfo = ({ info }) => (
                 </React.Fragment>
             }
             { info.nproc !== undefined &&
-                <tr>
-                    <th>{ _("CPU") }</th>
-                    <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
-                </tr>
+                <React.Fragment>
+                    <tr>
+                        <th>{ _("CPU") }</th>
+                        <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
+                    </tr>
+                    <tr>
+                        <th>{ _("CPU Security") }</th>
+                        <td>{ onSecurityClick !== undefined && <a role="link" onClick={onSecurityClick}> Mitigations </a> }</td>
+                    </tr>
+                </React.Fragment>
             }
         </tbody>
     </table>
 );
 
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+        };
+        this.nosmtRow();
+    }
+
+    close() {
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    nosmtRow() {
```

Comment:
This name is a bit confusing, it sounds like it was a part of `render()`. How about `detectNoSMT()`?